pkg/operator/controller/status/controller.go

```diff
@@ -176,44 +179,60 @@ func initializeClusterOperator(co *configv1.ClusterOperator) {
 }
 
 type operatorState struct {
-	Namespace *corev1.Namespace
-	DNS       operatorv1.DNS
+	haveNamespace bool
+	namespace     corev1.Namespace
+	haveDNS       bool
+	dns           operatorv1.DNS
 }
 
 // getOperatorState gets and returns the resources necessary to compute the
 // operator's current state.
-func (r *reconciler) getOperatorState(nsName string) (operatorState, error) {
-	state := operatorState{}
+func (r *reconciler) getOperatorState() (operatorState, error) {
+	var state operatorState
 
-	ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: nsName}}
-	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: nsName}, ns); err != nil {
+	ns := &corev1.Namespace{}
+	name := types.NamespacedName{
+		Name: operatorcontroller.DefaultOperandNamespace,
+	}
+	if err := r.client.Get(context.TODO(), name, ns); err != nil {
 		if !errors.IsNotFound(err) {
-			fmt.Printf("failed to get ns %s: %v\n", nsName, err)
-			return state, fmt.Errorf("failed to get namespace %q: %v", nsName, err)
+			return state, fmt.Errorf("failed to get namespace %q: %w", name, err)
 		}
 	} else {
-		state.Namespace = ns
+		state.haveNamespace = true
```

Comment:
I might be missing the point of this, but would it make sense to set `state.haveNamspace` to false after line 200?

Reply:
The zero value of `state.haveNamespace` is false.  I can add an assignment to reduce the risk of confusion.

Reply:
> The zero value of `state.haveNamespace` is false. I can add an assignment to reduce the risk of confusion.

Right. I was more so concerned about the scenario where the operand namespace is somehow deleted after `state.haveNamespace` is set to true (although ideally that would never happen, I suppose).

Reply:
`getOperatorState` declares a new `var state operatorState` every invocation, so it doesn't matter if a previous `getOperatorState` invocation returned `state` with `state.haveNamespace` set to true.